In [196]:
df1

person_age  person_education  person_income  person_emp_exp  \
0            22.0                 4        71948.0             0.0   
1            21.0                 3        12282.0             0.0   
2            25.0                 3        12438.0             3.0   
3            23.0                 1        79753.0             0.0   
4            24.0                 4        66135.0             1.0   
...           ...               ...            ...             ...   
44995        27.0                 0        47971.0             6.0   
44996        37.0                 0        65800.0            17.0   
44997        33.0                 0        56942.0             7.0   
44998        29.0                 1        33164.0             4.0   
44999        24.0                 3        51609.0             1.0   

       person_home_ownership  loan_amnt  loan_intent  loan_int_rate  \
0                          3  23093.125            4          16.02   
1                          2   1000.000            1          11.14   
2                          0   5500.000            3          12.87   
3                          3  23093.125            3          15.23   
4                          3  23093.125            3          14.27   
...                      ...        ...          ...            ...   
44995                      3  15000.000            3          15.66   
44996                      3   9000.000            2          14.07   
44997                      3   2771.000            0          10.02   
44998                      3  12000.000            1          13.23   
44999                      3   6665.000            0          17.05   

       loan_percent_income  cb_person_cred_hist_length  credit_score  \
0                     0.37                         3.0         561.0   
1                     0.08                         2.0         504.0   
2                     0.37                         3.0         635.0   
3                     0.37                         2.0         675.0   
4                     0.37                         4.0         586.0   
...                    ...                         ...           ...   
44995                 0.31                         3.0         645.0   
44996                 0.14                        11.0         621.0   
44997                 0.05                        10.0         668.0   
44998                 0.36                         6.0         604.0   
44999                 0.13                         3.0         628.0   

       loan_status  person_gender_male  previous_loan_defaults_on_file_Yes  
0                1                   0                                   0  
1                0                   0                                   1  
2                1                   0                                   0  
3                1                   0                                   0  
4                1                   1                                   0  
...            ...                 ...                                 ...  
44995            1                   1                                   0  
44996            1                   0                                   0  
44997            1                   1                                   0  
44998            1                   1                                   0  
44999            1                   1                                   0  

[45000 rows x 14 columns]

In [197]:
indep_X=df1.drop('loan_status', axis=1)
dep_Y=df1['loan_status']

In [225]:
df1.corr()

person_income  person_home_ownership  \
person_income                            1.000000              -0.373632   
person_home_ownership                   -0.373632               1.000000   
loan_amnt                                0.411985              -0.148551   
loan_int_rate                           -0.016266               0.130554   
loan_percent_income                     -0.357582               0.150731   
credit_score                             0.026481              -0.006358   
loan_status                             -0.249146               0.233842   
previous_loan_defaults_on_file_Yes       0.124726              -0.125974   

                                    loan_amnt  loan_int_rate  \
person_income                        0.411985      -0.016266   
person_home_ownership               -0.148551       0.130554   
loan_amnt                            1.000000       0.142671   
loan_int_rate                        0.142671       1.000000   
loan_percent_income                  0.612110       0.127736   
credit_score                         0.009183       0.011606   
loan_status                          0.107306       0.331851   
previous_loan_defaults_on_file_Yes  -0.058137      -0.181700   

                                    loan_percent_income  credit_score  \
person_income                                 -0.357582      0.026481   
person_home_ownership                          0.150731     -0.006358   
loan_amnt                                      0.612110      0.009183   
loan_int_rate                                  0.127736      0.011606   
loan_percent_income                            1.000000     -0.011436   
credit_score                                  -0.011436      1.000000   
loan_status                                    0.384660     -0.007680   
previous_loan_defaults_on_file_Yes            -0.202886     -0.183090   

                                    loan_status  \
person_income                         -0.249146   
person_home_ownership                  0.233842   
loan_amnt                              0.107306   
loan_int_rate                          0.331851   
loan_percent_income                    0.384660   
credit_score                          -0.007680   
loan_status                            1.000000   
previous_loan_defaults_on_file_Yes    -0.543096   

                                    previous_loan_defaults_on_file_Yes  
person_income                                                 0.124726  
person_home_ownership                                        -0.125974  
loan_amnt                                                    -0.058137  
loan_int_rate                                                -0.181700  
loan_percent_income                                          -0.202886  
credit_score                                                 -0.183090  
loan_status                                                  -0.543096  
previous_loan_defaults_on_file_Yes                            1.000000

# What is the relation between Loan Status and Previous Loan Defaults?

The correlation between loan_status and previous_loan_defaults_on_file_Yes is −0.543, indicating a strong negative correlation.
This suggests that applicants who have previously defaulted on loans are much less likely to have a positive loan status in the current application.

# What is the relation between Loan Status and Loan Percent Income?

The correlation between loan_status and loan_percent_income is 0.384, showing a moderate positive relationship.
This means that applicants who spend a larger percentage of their income on loan repayments are more likely to have unfavorable loan statuses, as higher repayment burdens increase the risk of default.

In [227]:
df1.cov()

person_income  person_home_ownership  \
person_income                        1.449460e+09          -20490.802426   
person_home_ownership               -2.049080e+04               2.075030   
loan_amnt                            9.148974e+07           -1248.177297   
loan_int_rate                       -1.843034e+03               0.559688   
loan_percent_income                 -1.148462e+03               0.018317   
credit_score                         5.020898e+04              -0.456134   
loan_status                         -3.943521e+03               0.140043   
previous_loan_defaults_on_file_Yes   2.373993e+03              -0.090722   

                                       loan_amnt  loan_int_rate  \
person_income                       9.148974e+07   -1843.033581   
person_home_ownership              -1.248177e+03       0.559688   
loan_amnt                           3.402331e+07    2476.656150   
loan_int_rate                       2.476656e+03       8.856988   
loan_percent_income                 3.012003e+02       0.032070   
credit_score                        2.667670e+03       1.720187   
loan_status                         2.602194e+02       0.410594   
previous_loan_defaults_on_file_Yes -1.695343e+02      -0.270345   

                                    loan_percent_income  credit_score  \
person_income                              -1148.462010  50208.979081   
person_home_ownership                          0.018317     -0.456134   
loan_amnt                                    301.200294   2667.669983   
loan_int_rate                                  0.032070      1.720187   
loan_percent_income                            0.007117     -0.048047   
credit_score                                  -0.048047   2480.174427   
loan_status                                    0.013491     -0.159015   
previous_loan_defaults_on_file_Yes            -0.008557     -4.558537   

                                    loan_status  \
person_income                      -3943.521262   
person_home_ownership                  0.140043   
loan_amnt                            260.219444   
loan_int_rate                          0.410594   
loan_percent_income                    0.013491   
credit_score                          -0.159015   
loan_status                            0.172843   
previous_loan_defaults_on_file_Yes    -0.112882   

                                    previous_loan_defaults_on_file_Yes  
person_income                                              2373.992781  
person_home_ownership                                        -0.090722  
loan_amnt                                                  -169.534250  
loan_int_rate                                                -0.270345  
loan_percent_income                                          -0.008557  
credit_score                                                 -4.558537  
loan_status                                                  -0.112882  
previous_loan_defaults_on_file_Yes                            0.249942

# What is the relation between Loan Amount and Credit Score?

The covariance between loan_amnt and credit_score is 2667.67, which is moderately positive.
Borrowers with higher credit scores tend to secure larger loan amounts, reflecting their financial credibility and strong credit history.

# What is the relation between Credit Score and Loan Status?

In [233]:
from scipy.stats import ttest_ind

group1 = df1[df1['loan_status'] == 0]['credit_score']
group2 = df1[df1['loan_status'] == 1]['credit_score']

t_stat, p_val = ttest_ind(group1, group2)
print("T-statistic:", t_stat)
print("P-value:", p_val)

T-statistic: 1.62921759338469
P-value: 0.10327396370091947


The p-value (0.103) is greater than 0.05, which means the difference in average credit scores between the two loan status groups (approved vs not approved) is not statistically significant.